pkg/engine/jmespath/functions_test.go

```diff
@@ -1381,3 +1382,113 @@ func Test_ObjectFromLists(t *testing.T) {
+	}
+	resExpected := make([]map[string]interface{}, 2)
+
+	for i, v := range resList {
+		err := json.Unmarshal([]byte(v), &resExpected[i])
+		assert.NilError(t, err)
+	}
+
+	testCases := []struct {
+		jmesPath       string
+		expectedResult map[string]interface{}
+	}{
+		{
+			jmesPath:       "x509_decode(base64_decode('" + certs[0] + "'))",
+			expectedResult: resExpected[0],
+		},
+		{
+			jmesPath:       "x509_decode(base64_decode('" + certs[1] + "'))",
+			expectedResult: resExpected[1],
+		},
+		{
+			jmesPath:       "x509_decode('" + certs[2] + "')",
+			expectedResult: resExpected[0],
+		},
+		{
+			jmesPath:       "x509_decode('" + certs[3] + "')",
+			expectedResult: resExpected[1],
+		},
+		{
+			jmesPath:       "x509_decode('" + certs[4] + "')",
+			expectedResult: resExpected[0],
+		},
+		{
+			jmesPath:       "x509_decode('" + certs[5] + "')",
+			expectedResult: resExpected[1],
+		},
+		{
+			jmesPath:       "x509_decode('xyz')",
+			expectedResult: map[string]interface{}{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.jmesPath, func(t *testing.T) {
+			jp, err := New(tc.jmesPath)
+			assert.NilError(t, err)
+
+			result, err := jp.Search("")
+			if err != nil && err.Error() != "invalid certificate" {
+				assert.NilError(t, err)
+			}
+
+			res, ok := result.(map[string]interface{})
+			assert.Assert(t, ok)
+
+			assert.Equal(t, true, cmp.Equal(res, tc.expectedResult))
```

Comment:
Can we remove the dependency on go-cmp? testify's assert.Equal should be capable of doing recursive equality comparison.

Reply:
Switched to `assert.DeepEqual` which also utilizes `go-cmp` internally, thanks for mentioning this!